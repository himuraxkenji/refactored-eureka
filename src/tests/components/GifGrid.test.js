import '@testing-library/jest-dom';
import React from 'react';
import {shallow} from 'enzyme';
import {GifGrid} from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el <GifGrid />', () => {

  const category = 'One Punch';
  // let wrapper = shallow(<GifGrid category={category} />);

  // beforeEach(() => {
  //   jest.clearAllMocks();
  //   wrapper = shallow(<GifGrid category={category} />)
  // })


  test('debe de mostrarse correctament <GifGrid />', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();

  });


  test('debe de mostrar items cuando se cargan imagenes use useFetchGifs', () => {

    const gifs = [{
      id: 'url',
      url: 'https://false.jpg',
      title: 'Cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });


    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  })

});
