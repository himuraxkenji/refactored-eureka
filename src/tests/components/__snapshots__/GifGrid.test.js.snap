// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrar items cuando se cargan imagenes use useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="url"
      key="url"
      title="Cualquier cosa"
      url="https://false.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> debe de mostrarse correctament <GifGrid /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
